cdc/capture/http_handler.go

```diff
@@ -0,0 +1,635 @@
+			return
+		}
+
+		resp := &model.ChangefeedCommonInfo{
+			ID: changefeedID,
+		}
+
+		if cfInfo != nil {
+			resp.FeedState = cfInfo.State
+			resp.RunningError = cfInfo.Error
+		}
+
+		if cfStatus != nil {
+			resp.CheckpointTSO = cfStatus.CheckpointTs
+			tm := oracle.GetTimeFromTS(cfStatus.CheckpointTs)
+			resp.CheckpointTime = model.JSONTime(tm)
+		}
+
+		resps = append(resps, resp)
+	}
+	c.IndentedJSON(http.StatusOK, resps)
+}
+
+// GetChangefeed get detailed info of a changefeed
+// @Summary Get changefeed
+// @Description get detail information of a changefeed
+// @Tags changefeed
+// @Accept  json
+// @Produce  json
+// @Param   changefeed_id  path  string  true  "changefeed_id"
+// @Success 200 {object} model.ChangefeedDetail
+// @Failure 500,400 {object} model.HTTPError
+// @Router	/api/v1/changefeeds/{changefeed_id} [get]
+func (h *HTTPHandler) GetChangefeed(c *gin.Context) {
+	changefeedID := c.Param(APIOpVarChangefeedID)
+	if err := model.ValidateChangefeedID(changefeedID); err != nil {
+		c.IndentedJSON(http.StatusBadRequest,
+			model.NewHTTPError(cerror.ErrAPIInvalidParam.GenWithStack("invalid changefeed_id: %s", changefeedID)))
+		return
+	}
+
+	info, err := h.capture.etcdClient.GetChangeFeedInfo(c, changefeedID)
+	if err != nil {
+		c.IndentedJSON(http.StatusInternalServerError, model.NewHTTPError(err))
+		return
+	}
+
+	status, _, err := h.capture.etcdClient.GetChangeFeedStatus(c, changefeedID)
+	if err != nil {
+		c.IndentedJSON(http.StatusInternalServerError, model.NewHTTPError(err))
+		return
+	}
+
+	processorInfos, err := h.capture.etcdClient.GetAllTaskStatus(c, changefeedID)
+	if err != nil {
+		c.IndentedJSON(http.StatusInternalServerError, model.NewHTTPError(err))
+		return
+	}
+
+	taskStatus := make([]model.CaptureTaskStatus, 0, len(processorInfos))
+	for captureID, status := range processorInfos {
+		tables := make([]int64, 0)
+		for tableID := range status.Tables {
+			tables = append(tables, tableID)
+		}
+		taskStatus = append(taskStatus, model.CaptureTaskStatus{CaptureID: captureID, Tables: tables, Operation: status.Operation})
+	}
+
+	changefeedDetail := &model.ChangefeedDetail{
+		ID:             changefeedID,
+		SinkURI:        info.SinkURI,
+		CreateTime:     model.JSONTime(info.CreateTime),
+		StartTs:        info.StartTs,
+		TargetTs:       info.TargetTs,
+		CheckpointTSO:  status.CheckpointTs,
+		CheckpointTime: model.JSONTime(oracle.GetTimeFromTS(status.CheckpointTs)),
+		Engine:         info.Engine,
+		FeedState:      info.State,
+		TaskStatus:     taskStatus,
+	}
+
+	c.IndentedJSON(http.StatusOK, changefeedDetail)
+}
+
+// CreateChangefeed creates a changefeed
+// @Summary Create changefeed
+// @Description create a new changefeed
+// @Tags changefeed
+// @Accept  json
+// @Produce  json
+// @Param  changefeed body model.ChangefeedConfig true "changefeed config"
+// @Success 202
+// @Failure 500,400 {object} model.HTTPError
+// @Router	/api/v1/changefeeds [post]
```

Comment:
```suggestion
// @Tags changefeed
// @Accept json
// @Produce json
// @Param changefeed body model.ChangefeedConfig true "changefeed config"
// @Success 202
// @Failure 500,400 {object} model.HTTPError
// @Router /api/v1/changefeeds [post]
```

Please use one space ` `.

Reply:
done